### PR TITLE
ECR support (Based on drone-plugins/drone-docker)

### PR DESCRIPTION
**Pull request by johnrengelman**
the `.drone.sec` file in this repo was copied from https://github.com/drone-plugins/drone-docker/blob/master/.drone.sec

### CODE REVIEW

File: DOCS.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    * `tag` - cherry-pick tags to save (optional)
+* `load` - restore image layers from the specified tar file
+* `build_args` - [build arguments](https://docs.docker.com/engine/reference/commandline/build/#set-build-time-variables-build-arg) to pass to `docker build`
+
+The following is a sample Docker configuration in your .drone.yml file:
+
+```yaml
+publish:
+  ecr:
+    accessKey: MyAWSAccessKey
```
Comment (tboerger) [inline DOCS.md]: that should be snakecase

Comment (johnrengelman) [inline DOCS.md]: Pushed fix.

**Comment by johnrengelman** [comment]
Any other comments here? I really need to get this into place.

Also not this plugin will require explicit white listing here: https://github.com/drone/drone-exec/blob/8eb0fdf695c6e6870c1bc48f4414c04d4b33a47a/runner/build.go#L124 and here: https://github.com/drone/drone-exec/blob/970e556f6027b4663b6b0c98f6705da8bd8e4749/parser/funcs.go#L148

**Comment by jackspirou** [comment]
LGTM

**Comment by johnrengelman** [comment]
Added support in `drone-exec`: https://github.com/drone/drone-exec/pull/19

**Comment by johnrengelman** [comment]
@bradrydzewski is this good? Can CI be set up for it?

**Comment by tboerger** [comment]
The documentation is really a hard copy of the docker plugin... That should be customized a little bit to really match this plugin, it's simply not the drone-docker plugin anymore :)

**Comment by johnrengelman** [comment]
@tboerger it's exactly the same except it has the extra flow of making an AWS call to get a username/password before logging into the registry.

I'll add some specifics about that...but other than the access_key/secret_key variables, this is a drop in replacement from the drocker plugin.

**Comment by bradrydzewski** [comment]
I'll setup the CI today

**Pull request merged**